content/docs/hooks-recipes.md

```diff
@@ -10,3 +10,157 @@ prev: hooks-faq.html
 This is a collection of recipes that may help you start "thinking in Hooks". Feel free to play with them or copy and paste them into your project or tweak them to your needs.
 
 ## Data Fetching
+
+## Subscriptions
+
+Sometimes React components need to subscribe to an external data source. The `useSubscription` hook below shows how to do this in a way that is compatibile with React's upcoming concurrent rendering mode.
+
+> Note
+>
+> Subscriptions may not be the best solution for your use case. For example:
+> * Redux/Flux stores should use the [context API](/docs/context.html) instead.
+> * Complex libraries like Relay/Apollo should manage subscriptions manually with the same technique demonstrated below.
+> * In the future, I/O subscriptions (e.g. notifications) that update infrequently should use [`react-cache`](https://github.com/facebook/react/blob/master/packages/react-cache/README.md).
```

Comment:
We should have no mention of react-cache in official doc. We might not even keep the name. It's also confusing to read about the future in a recipes section — people want to do something today.